src/SpritesheetBuilder.js

```diff
@@ -0,0 +1,134 @@
+    this.cache[moveKey].src = EMPTY_PNG;
+
+    this.moveQueue.push([moveKey, character, move, moveFrames]);
+    if (this.moveQueue.length === 1) {
+      this.startQueue();
+    }
+  }
+
+  startQueue() {
+    this.p5._incrementPreload();
+    this.onStop = () => this.p5._decrementPreload();
+
+    this.cachedSvgImg = new Image();
+    this.canvas = document.createElement('canvas');
+    this.canvas.width = CACHED_SIZE * 6 / 2;
+    this.canvas.height = CACHED_SIZE * 4;
+    this.context = this.canvas.getContext('2d');
+    setTimeout(this.processQueue, 0);
+  }
+
+  processQueue() {
+    if (this.moveQueue.length <= 0) {
+      console.log('done');
+      delete this.context;
+      delete this.canvas;
+      delete this.cachedSvgImg;
+      if (typeof this.onStop === 'function') {
+        this.onStop();
+      }
+      return;
+    }
+
+    this.rasterizeMove(this.moveQueue[0], () => {
+      this.moveQueue.shift();
+      setTimeout(this.processQueue, 0);
+    });
+  }
+
+  rasterizeMove(moveData, callback) {
+    const [moveKey, character, move, moveFrames] = moveData;
+    // const moveName = window.nativeAPI.world.MOVE_NAMES[move].name;
+    // const svgUrl = `assets/${character.toLowerCase()}.svg#${moveName}`;
+    const svgUrl = `assets/${character.toLowerCase()}.svg`;
+
+    const rasterize = () => {
+      console.count('rasterize');
+      this.context.clearRect(0, 0, CACHED_SIZE * 6, CACHED_SIZE * 4);
+      for (let frame = 0; frame < moveFrames; frame++) {
+        let row = Math.floor(frame / 6);
+        let col = frame % 6;
+        this.context.drawImage(this.cachedSvgImg,
+          (frame + 0.25) * SOURCE_SIZE, move * SOURCE_SIZE, SOURCE_SIZE / 2, SOURCE_SIZE,
+          col * CACHED_SIZE / 2, row * CACHED_SIZE, CACHED_SIZE / 2, CACHED_SIZE
+        );
+      }
+      // this.context.drawImage(this.cachedSvgImg, 0, move * SOURCE_SIZE, SOURCE_ROW_SIZE, SOURCE_SIZE, 0, 0, CACHED_ROW_SIZE, CACHED_SIZE);
+      // this.context.drawImage(this.cachedSvgImg, SOURCE_ROW_SIZE, move * SOURCE_SIZE, SOURCE_ROW_SIZE, SOURCE_SIZE, 0, CACHED_SIZE, CACHED_ROW_SIZE, CACHED_SIZE);
+      // if (moveFrames > 12) {
+      //   this.context.drawImage(this.cachedSvgImg, 2 * SOURCE_ROW_SIZE, move * SOURCE_SIZE, SOURCE_ROW_SIZE, SOURCE_SIZE, 0, 2 * CACHED_SIZE, CACHED_ROW_SIZE, CACHED_SIZE);
+      //   this.context.drawImage(this.cachedSvgImg, 3 * SOURCE_ROW_SIZE, move * SOURCE_SIZE, SOURCE_ROW_SIZE, SOURCE_SIZE, 0, 3 * CACHED_SIZE, CACHED_ROW_SIZE, CACHED_SIZE);
+      // }
+      this.cache[moveKey].src = this.canvas.toDataURL();
+      callback();
+    };
+
+    const skip = () => {
+      console.log(`Couldn't load '${svgUrl}'. Skipping...`);
+      callback();
+    };
+
+    if (this.cachedSvgImg.src !== svgUrl) {
+      this.cachedSvgImg.onload = rasterize;
+      this.cachedSvgImg.onerror = skip;
+      this.cachedSvgImg.src = svgUrl;
+    } else {
+      rasterize();
+    }
+  }
+}
+
+module.exports = SpritesheetBuilder;
```

Comment:
Some general feedback: I'm not sure it makes sense for this to be a class. Seems we might instead be able to just export a `getSpriteSheet`.

If we do want to have this as a class, it might make sense to move local functions like `rasterize` to also be class functions. I found the logic flows here slightly difficult to parse in general, and I think one thing I struggled with was why some functions were defined locally vs. on the class.

I think another thing that would make a lot of this easier to follow would be sticking some param/return js-doc tags on methods.